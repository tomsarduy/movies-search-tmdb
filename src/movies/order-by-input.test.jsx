import React from "react";
import OrderByInput from "./order-by-input.jsx";

describe("OrderByInput", () => {
  it("renders correctly", () => {
    const component = shallow(<OrderByInput onChange={() => null} />);
    expect(component).toMatchSnapshot();
  });

  it("should call onChange with the selected value on change", () => {
    const onChangeMock = jest.fn();
    const target = { value: "title" };
    const component = shallow(<OrderByInput onChange={onChangeMock} />);
    component.find("select").simulate("change", { target });
    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toHaveBeenCalledWith({ target });

    // Check with another value
    component
      .find("select")
      .simulate("change", { target: { value: "popularity" } });
    expect(onChangeMock).toBeCalledTimes(2);
    expect(onChangeMock).toHaveBeenCalledWith({
      target: { value: "popularity" }
    });
  });
});
