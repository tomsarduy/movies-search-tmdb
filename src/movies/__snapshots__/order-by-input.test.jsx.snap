// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderByInput renders correctly 1`] = `
<div>
  <span>
    <b>
      Sort by: 
    </b>
  </span>
  <select
    defaultValue="title"
    id="sort-by"
    name="sort-by"
    onChange={[Function]}
  >
    <option
      value="title"
    >
      Title
    </option>
    <option
      value="popularity"
    >
      Popularity
    </option>
    <option
      value="release_date"
    >
      Release Date
    </option>
  </select>
</div>
`;
